### PR TITLE
Use only BIT* defines in Buffer

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System/Buffer.cs [hunk 1]
```diff
@@ -2,7 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#if AMD64 || ARM64 || (BIT32 && !ARM)
+#if BIT64 || (BIT32 && !ARM)
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Buffer.cs]: `HAS_CUSTOM_BLOCKS` happens to work well for the two 64-bit platforms that are supported currently, but is it guaranteed to be the case for all 64-bit platforms?

Comment (marek-safar) [reply src/System.Private.CoreLib/shared/System/Buffer.cs]: it's not guaranteed but it's the same as the rest of the condition. can you guarantee it works for all 32-bit platforms except ARM?

Comment (jkotas) [reply src/System.Private.CoreLib/shared/System/Buffer.cs]: The conditions like these typically need to be tweaked as new platforms are brought online. If necessary, it is fine to tweak this condition to make it work better for platforms supported by Mono.

I think it is useful to have the path that copies the memory as 64-bit integers around for performance comparison experiments and for potentially being used by platforms that are not supported yet.

Comment (marek-safar) [reply src/System.Private.CoreLib/shared/System/Buffer.cs]: Fair enough, ARM64 is too broad anyway so will tweak that as we go

**Pull request closed without merging**